tl/features/cell_context_matches.py

```diff
@@ -213,16 +218,97 @@ def initialize(self, input_df, context_dict, label_column):
                                            score=context_result['score']
                                            )
 
+        property_val_df, important_properties = self.compute_property_scores(row_col_label_dict, n_cntxt_columns, num_rows)
+        context_score_list = []
+        context_property_list = []
+        context_similarity_list = []
+        for row, col, q_node in zip(input_df['row'], input_df['column'], input_df['kg_id']):
+            # Handle equal similarity for different properties by looping over and getting the one with highest similarity.
+            context_score = 0.0
+            property_matched = []
+            similarity_matched = []
+            r_c = f"{row}_{col}"
+            for cols in range(n_cntxt_columns):
+                if int(cols) != int(col):
+                    returned_properties = self.ccm_dict[r_c].get_properties(cols, q_node_dependent=True, q_node=q_node)
+                    if returned_properties == []:
+                        continue
+                    (property_, type, best_score, _) = returned_properties[0]
+                    property_value = property_val_df.loc[(property_val_df['property_'] == property_) & (property_val_df['column'] == col) & (property_val_df['col_2'] == cols), 'property_score']
+                    property_value = round(property_value, 4)
+                    context_score = context_score + (property_value * best_score)
+                    property_matched.append(property_ + str(property_value))
+                    similarity_matched.append(best_score)
+            context_score_list.append(context_score)
+            context_similarity_list.append(similarity_matched)
+            context_property_list.append(property_matched)
+        # input_df['context_property'] = context_property_list
+        # input_df['context_similarity'] = context_similarity_list
+        input_df['context_score'] = context_score_list
+
+    def compute_property_scores(self, row_col_label_dict, n_cntxt_columns, num_rows):
+        # To calculate property score
+        properties_df = pd.DataFrame()
+        for r_c in row_col_label_dict:
+            row_col = r_c.split("_")
+            row_1 = row_col[0]
+            col_1 = row_col[1]
+            for cols in range(n_cntxt_columns):
+                if int(cols) != int(col_1):
+                    m = self.ccm_dict[r_c].get_properties(cols, q_node_dependent=True)
+                    int_prop = pd.DataFrame(m, columns = ["property_", "type", "best_score", "n_occurences"])
+                    int_prop['row'] = int(row_1)
+                    int_prop['column'] = int(col_1)
+                    int_prop['col_2'] = int(cols)
```

Comment:
follow the convention, `col_2` should be `col2`

Reply:
corrected.